internal/functions/efgs/database/database.go

```diff
@@ -85,7 +85,7 @@ func init() {
 	}
 }
 
-//PersistDiagnosisKeys Save array of DiagnosisKey to database
+//PersistDiagnosisKeys Save array of DiagnosisKey to database. If there is any duplication, no keys will be saved.
```

Comment:
So, if any device tries to upload keys that are already uploaded, the whole set is rejected, including new keys? **Maybe** good-enough for production but sounds terrible for testing :-)